web-registry/src/hooks/projects/useProjectWithMetadata.ts

```diff
@@ -0,0 +1,167 @@
+import { useCallback } from 'react';
+import { ProjectInfo } from '@regen-network/api/lib/generated/regen/ecocredit/v1/query';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+
+import {
+  ProjectByIdQuery,
+  ProjectByOnChainIdQuery,
+  useUpdateProjectByIdMutation,
+} from 'generated/graphql';
+import { graphqlClient } from 'lib/clients/graphqlClient';
+import { getProjectKey } from 'lib/queries/react-query/ecocredit/getProjectQuery/getProjectQuery.constants';
+import { getMetadataQuery } from 'lib/queries/react-query/registry-server/getMetadataQuery/getMetadataQuery';
+import { getProjectByIdQuery } from 'lib/queries/react-query/registry-server/graphql/getProjectByIdQuery/getProjectByIdQuery';
+import { getProjectByIdKey } from 'lib/queries/react-query/registry-server/graphql/getProjectByIdQuery/getProjectByIdQuery.constants';
+import { getProjectByOnChainIdQuery } from 'lib/queries/react-query/registry-server/graphql/getProjectByOnChainIdQuery/getProjectByOnChainIdQuery';
+import { getProjectByOnChainIdKey } from 'lib/queries/react-query/registry-server/graphql/getProjectByOnChainIdQuery/getProjectByOnChainIdQuery.constants';
+
+import { ReturnType as ProjectEditSubmitReturnType } from 'pages/ProjectEdit/hooks/useProjectEditSubmit';
+import {
+  BasicInfoFormValues,
+  DescriptionValues,
+  ProjectLocationFormValues,
+} from 'components/organisms';
+import { MediaValues } from 'components/organisms/MediaForm';
+
+export type OffChainProject =
+  | ProjectByIdQuery['projectById']
+  | ProjectByOnChainIdQuery['projectByOnChainId'];
+
+interface Props {
+  projectId?: string;
+  isEdit?: boolean;
+  projectEditSubmit: ProjectEditSubmitReturnType;
+  navigateNext: () => void;
+  anchored?: boolean;
+  onChainProject?: ProjectInfo;
+}
+
+interface Res {
+  offChainProject?: OffChainProject;
+  metadata: any; // TODO update with proper type
+  metadataReload: () => Promise<void>;
+  metadataSubmit: (values: Values) => Promise<void>;
+}
+
+type Values =
+  | BasicInfoFormValues
+  | ProjectLocationFormValues
+  | DescriptionValues
+  | MediaValues;
+
+export const useProjectWithMetadata = ({
+  projectId,
+  isEdit,
+  onChainProject,
+  navigateNext,
+  projectEditSubmit,
+  anchored = true,
+}: Props): Res => {
+  let metadata: any;
+  let offChainProject: OffChainProject | undefined;
+  const reactQueryClient = useQueryClient();
+  const [updateProject] = useUpdateProjectByIdMutation();
+
+  // In project creation mode, we query the off-chain project since there's no on-chain project yet.
+  // In this case, the router param projectId is the off-chain project uuid.
+  const create = !!projectId && !isEdit;
+  const { data: projectByIdRes } = useQuery(
```

Comment:
```suggestion
  const { data: projectByOffChainIdRes } = useQuery(
```
Would that make sense ?